### PR TITLE
Revert to use standard library definitions in API functions and examples

### PR DESCRIPTION
**Pull request by IshaanDesai**
## Description

This PR reverts to using `std::string` and `std::vector` in the adapter API functions. During the update for compatibility to `precice:develop`, inadvertently the API input variables were modified to be of the type `precice::string_view` and `precice::span`, which is actually not intended, as these types are not to be used directly by the user.

## Checklist

- [x] I made sure that the source files are formatted properly.
- [ ] I added my changes to the changelog (`CHANGELOG.md`)

### CODE REVIEW

File: dumux-precice/couplingadapter.cc [hunk 1]
```diff
@@ -48,15 +48,14 @@ void CouplingAdapter::announceQuantity(const precice::string_view &meshName,
...
+int CouplingAdapter::getMeshDimensions(const std::string &meshName) const
 {
     assert(wasCreated_);
     return precice_->getMeshDimensions(meshName);
 }
 
-void CouplingAdapter::setMesh(const precice::string_view &meshName,
-                              precice::span<const double> positions)
+void CouplingAdapter::setMesh(const std::string &meshName,
+                              std::vector<double> &positions)
```
Comment (mathiskelm) [inline dumux-precice/couplingadapter.cc:58]: Do we want to stick to const positions? Could also make other data vectors and strings const.
```suggestion
                              const std::vector<double> &positions)
```

File: examples/dummysolver/main_dummysolver.cc [hunk 2]
```diff
@@ -42,7 +42,7 @@ try {
         getParamFromGroup<std::string>("preCICE", "ConfigFileName");
     const std::string meshName =
         getParamFromGroup<std::string>("preCICE", "MeshName");
-    const precice::string_view meshNameView(meshName);
+    const std::string meshNameView(meshName);
```
Comment (mathiskelm) [inline examples/dummysolver/main_dummysolver.cc:45]: We don't need a separate view of strings anymore, do we? Should be able to replace all `meshNameView` etc. with the string versions.

File: examples/dummysolver/main_dummysolver.cc [hunk 3]
```diff
@@ -54,13 +54,13 @@ try {
 
     const int dimensions = couplingParticipant.getMeshDimensions(meshNameView);
     assert(dimensions == 3);
-    const precice::string_view scalarDataWriteName = std::string(
+    const std::string scalarDataWriteName = std::string(
         (solverName == "SolverOne") ? "scalarDataOne" : "scalarDataTwo");
```
Comment (mathiskelm) [inline examples/dummysolver/main_dummysolver.cc:58]: More below
```suggestion
    const std::string scalarDataWriteName = 
        (solverName == "SolverOne") ? "scalarDataOne" : "scalarDataTwo";
```

File: examples/dummysolver/main_dummysolver.cc (hunk 2 above)
Comment (IshaanDesai) [reply examples/dummysolver/main_dummysolver.cc:45]: We could also just pass literals in every preCICE API call.

File: dumux-precice/couplingadapter.cc [hunk 4]
```diff
@@ -156,16 +153,16 @@ void CouplingAdapter::writeScalarQuantityOnFace(
...
     std::string key = meshAndDataKey(meshName, dataName);
     assert(dataMap_.find(key) != dataMap_.end());
     return dataMap_[key];
 }
 
-void CouplingAdapter::writeQuantityVector(const precice::string_view &meshName,
-                                          const precice::string_view &dataName,
+void CouplingAdapter::writeQuantityVector(const std::string &meshName,
+                                          const std::string &dataName,
                                           std::vector<double> &values)
```
Comment (mathiskelm) [inline dumux-precice/couplingadapter.cc:166]:
```suggestion
                                          const std::vector<double> &values)
```

File: dumux-precice/couplingadapter.hh [hunk 5]
```diff
@@ -242,18 +242,17 @@ public:
...
      *
      * @param[in] meshName Name of the mesh.
      * @param[in] dataName Name of the data.
      * @param[in] values Value of the scalar or vector quantity.
      */
-    void writeQuantityVector(const precice::string_view &meshName,
-                             const precice::string_view &dataName,
+    void writeQuantityVector(const std::string &meshName,
+                             const std::string &dataName,
                              std::vector<double> &values);
```
Comment (mathiskelm) [inline dumux-precice/couplingadapter.hh:256]:
```suggestion
                             const std::vector<double> &values);
```

File: examples/ff-pm/flow-over-cube-3d/main_ff-reversed.cc [hunk 6]
```diff
@@ -99,8 +99,8 @@ template<class FluxVariables,
 void setInterfacePressures(const Problem &problem,
                            const GridVariables &gridVars,
                            const SolutionVector &sol,
-                           const precice::string_view meshNameView,
-                           const precice::string_view dataNameView)
+                           const std::string meshName,
+                           const std::string dataNameView)
```
Comment (mathiskelm) [inline examples/ff-pm/flow-over-cube-3d/main_ff-reversed.cc:103]: other instances in this file
```suggestion
                           const std::string dataName)
```

File: examples/ff-pm/flow-over-cube-3d/main_pm-reversed.cc [hunk 7]
```diff
@@ -269,7 +269,7 @@ std::tuple<double, double, double> writeVelocitiesOnInterfaceToFile(
 }
 
 template<class Problem, class GridVariables, class SolutionVector>
-void writePressuresOnInterfaceToFile(const precice::string_view &meshName,
+void writePressuresOnInterfaceToFile(const std::string &meshName,
                                      std::string &filename,
```
Comment (mathiskelm) [inline examples/ff-pm/flow-over-cube-3d/main_pm-reversed.cc:273]:
```suggestion
                                     const std::string &filename,
```

**Review by mathiskelm — APPROVED** [review]

**Pull request merged**